### PR TITLE
add recipe for gumshoe

### PR DESCRIPTION
**Pull request by Overdr0ne**
### Brief summary of what the package does

Gumshoe is yet another point movement tracker. It primarily tracks point as a function of space, but will also log point movements when an idle time is exceeded. Each global minor mode keeps a movement log local to some scope: global, perspective or buffer, and each operates independently of each other, and the global mark ring. More scopes could be added, as long as they conform to the standard make-variable-*-local interface.

### Direct link to the package repository

https://github.com/Overdr0ne/gumshoe

### Your association with the package

I am the creator/maintainer.

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

### CODE REVIEW

**Comment by alphapapa** [comment]
This looks really cool (and nice picture in the readme!).  I'd suggest using https://github.com/alphapapa/makem.sh to lint your code.  Here's the output I get:

```
$ ./makem.sh -s lint

In end of data:
gumshoe.el:319:1:Warning: the following functions are not known to be defined:
    incf, decf, persp-make-variable-persp-local
ERROR (2021-07-29 23:27:55): Linting compilation failed.
Entering directory '/tmp/gumshoe/gumshoe/'
gumshoe.el:1:75: warning: You should depend on (emacs "24.1") if you need lexical-binding.
gumshoe.el:36:10: error: You should depend on (emacs "24.3") or the cl-lib package if you need `cl-lib'.
gumshoe.el:97:1: error: You should depend on (emacs "25.1") or the cl-lib package if you need `cl-defmethod'.
gumshoe.el:134:1: error: You should depend on (emacs "25.1") or the cl-lib package if you need `cl-defmethod'.
gumshoe.el:144:1: error: You should depend on (emacs "25.1") or the cl-lib package if you need `cl-defmethod'.
gumshoe.el:290:1: error: You should depend on (emacs "24.3") if you need `defvar-local'.
ERROR (2021-07-29 23:27:57): Linting package failed.
LOG (2021-07-29 23:27:57): Finished with 2 errors.
```

**Comment by Overdr0ne** [comment]
Thanks for getting back so quickly. I actually did see those. Maybe I'm misunderstanding, but I have `(require 'cl-lib)`, so doesn't that mean I "depend on (emacs "25.1") or the cl-lib package". Similarly, incf is a function alias for cl-incf. Is it the preferred style to use the cl- versions?

As for the persp-make-variable-persp-local, I only conditionally require prespective in gumshoe--global-gumshoe-persp-mode-enable called when global-gumshoe-persp-mode is enabled, and that's the only way that function should be called. Does that sound reasonable?

**Comment by alphapapa** [comment]
> Thanks for getting back so quickly. I actually did see those. Maybe I'm misunderstanding, but I have `(require 'cl-lib)`, so doesn't that mean I "depend on (emacs "25.1") or the cl-lib package"? 

You should generally declare the minimum Emacs version your package supports, anyway.

> Similarly, incf is a function alias for cl-incf. Is it the preferred style to use the cl- versions?

Yes.

> As for the persp-make-variable-persp-local, I only conditionally require prespective in gumshoe--global-gumshoe-persp-mode-enable called when global-gumshoe-persp-mode is enabled, and that's the only way that function should be called. Does that sound reasonable?

Seems okay to me.  You can use `declare-function` to silence the warning.

**Comment by Overdr0ne** [comment]
Everything should be fixed now:
https://github.com/Overdr0ne/gumshoe/commit/ab650e0ba17c7b0aad1a55a838533697ef418f42
https://github.com/Overdr0ne/gumshoe/commit/98a0ecaa89fb05a4a2c860fb101f5a61d1ffda47
https://github.com/Overdr0ne/gumshoe/commit/e0d9759cb31cd39a10a6bf61e362ac1d59f4ccd2

File: recipes/gumshoe [hunk 1]
```diff
@@ -0,0 +1,4 @@
+(gumshoe
+ :fetcher github
+ :repo "Overdr0ne/gumshoe"
+ :files ("gumshoe.el"))
```
Comment (riscy) [inline recipes/gumshoe:4]:
```suggestion
 :repo "Overdr0ne/gumshoe")
```

**Review by riscy — COMMENTED** [review]
Thanks both!  Checklist looks great now.

The "default" recipe looks like it is equivalent, and generally that's encouraged since it will give your package room to grow without subsequent PRs.

You should also add a standard GPL "LICENSE" file to your GitHub repository -- see: https://github.com/licensee/licensee

**Comment by Overdr0ne** [comment]
Sounds good, I've added a standard GPL license file:
https://github.com/Overdr0ne/gumshoe/commit/738ce468a84f55e0917a428cadfafc363adc8685

**Comment by riscy** [comment]
Thanks!

**Comment by Overdr0ne** [comment]
Thank you!

**Pull request merged**